Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs

```diff
@@ -26,144 +23,125 @@ namespace ECS.SceneLifeCycle.Systems
+                // waste time on virtual calls when accessing its elements.
+                DistanceBuckets = (List<int>)partitionSettings.SqrDistanceBuckets,
             };
-            partitionDataContainer.SetPartitionData(partitionData);
-        }
 
-        protected void AddCorners(ref SceneDefinitionComponent definition)
-        {
-            var corners = new NativeArray<ParcelCorners>(definition.ParcelsCorners.Count, Allocator.Persistent);
+            Profiler.BeginSample(nameof(PartitionScenes));
 
-            for (var i = 0; i < definition.ParcelsCorners.Count; i++)
-                corners[i] = definition.ParcelsCorners[i];
+            World.InlineQuery<PartitionScenes, SceneDefinitionComponent, PartitionComponent>(
+                in existingScenesQuery, ref partitionScenes);
 
-            partitionDataContainer.AddCorners(new ParcelCornersData(in corners));
-            definition.InternalJobIndex = partitionDataContainer.CurrentPartitionIndex;
+            Profiler.EndSample();
         }
 
-        [Query]
-        [None(typeof(PortableExperienceComponent))]
-        private void PartitionExistingEntity(ref SceneDefinitionComponent definition, ref PartitionComponent partitionComponent)
+        private struct PartitionScenes : IForEach<SceneDefinitionComponent, PartitionComponent>
         {
-            if (definition.InternalJobIndex < 0) return;
-            partitionDataContainer.SetPartitionComponentData(definition.InternalJobIndex, ref partitionComponent);
+            public Vector3 CameraPosition;
+            public Vector3 CameraForward;
+            public bool CameraIsDirty;
+            public List<int> DistanceBuckets;
+            public IComponentPool<PartitionComponent> PartitionPool;
+            public float UnloadingSqrDistance;
+
+            public void Update(ref SceneDefinitionComponent definition, ref PartitionComponent partition)
+            {
+                if (partition != null && !CameraIsDirty)
+                    return;
+                else if (partition == null)
+                    partition = PartitionPool.Get();
+
+                if (definition.IsPortableExperience)
+                {
+                    partition.OutOfRange = false;
+                    partition.Bucket = 0;
+                    partition.IsBehind = false;
+                    partition.RawSqrDistance = 0f;
+                    partition.IsDirty = true;
+                    return;
+                }
+
+                ParcelMathHelper.SceneCircumscribedPlanes sceneBounds2d = definition.SceneGeometry.CircumscribedPlanes;
+                Vector3 min = new (sceneBounds2d.MinX, 0f, sceneBounds2d.MinZ);
+                Vector3 max = new (sceneBounds2d.MaxX, definition.SceneGeometry.Height, sceneBounds2d.MaxZ);
+
+                Bounds sceneBounds = new ();
+                sceneBounds.SetMinMax(min, max);
+
+                float sqrDistance = sceneBounds.SqrDistance(CameraPosition);
```

Comment:
Behaviour change: measure the distance to the surface of the bounding box instead the distance to the nearest corner. This is more correct, but might break some other code due to accidental reliance on the quirks of the previous implementation.

Reply:
Behavior change: this implementation takes maximum scene height into account. So, if you fly high above the city, it will get unloaded. Or if you look up at the sky, a tall scene next to you will now not count as behind you.

Reply:
Behaviour change: use the bounds of the scene instead of the individual parcels to do the math. This is faster, but if the scene is oddly shaped, less accurate.